include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -452,7 +452,7 @@ struct __global_scan_functor
         constexpr auto __shift = _Inclusive{} ? 0 : 1;
         auto __item_idx = __item.get_linear_id();
         // skip the first group scanned locally
-        if (__item_idx >= __size_per_wg && __item_idx < __n)
+        if (__item_idx >= __size_per_wg && __item_idx < __n - __shift)
```

Comment:
Let's align with the code inside the condition:
```suggestion
        if (__item_idx >= __size_per_wg && __item_idx + __shift < __n)
```